Microsoft.Azure.Cosmos/src/Query/v3Query/QueryDefinition.cs

```diff
@@ -89,6 +89,58 @@ public QueryDefinition WithParameter(string name, object value)
             return this;
         }
 
+        /// <summary>
+        /// Checks for equality of two QueryDefinitions ignoring order of sqlParameters
+        /// </summary>
+        /// <param name="other"></param>
+        /// <returns>Boolean representing the equality</returns>
+        public bool Equals(QueryDefinition other)
+        {
+            {
+                if (this.QueryText != other.QueryText)
+                {
+                    return false;
+                }
+
+                if (this.SqlParameters.Count != other.SqlParameters.Count)
+                {
+                    return false;
+                }
+
+                foreach (KeyValuePair<string, SqlParameter> queryParameter in this.SqlParameters)
+                {
+                    if (!other.SqlParameters.TryGetValue(queryParameter.Key, out SqlParameter value) ||
+                        !value.Value.Equals(queryParameter.Value.Value) ||
+                        !value.Name.Equals(queryParameter.Value.Name))
+                    {
+                        return false;
+                    }
+                }
+
+                return true;
+            }
+        }
+
+        /// <summary>
+        /// Caculates a hashcode of QueryDefinition, ignoring order of sqlParameters.
+        /// </summary>
+        /// <returns>Hash code of the object.</returns>
+        public override int GetHashCode()
+        {
+            unchecked
+            {
+                int hashCode = 17;
+                foreach (KeyValuePair<string, SqlParameter> queryParameter in this.SqlParameters)
+                {
+                    int kvpHashCode = 17 + (queryParameter.Value.Name.GetHashCode() * 233) + queryParameter.Value.Value.GetHashCode();
```

Comment:
Are we missing the queryParameter.Key.GetHashCode()?

Reply:
Can we assume that queryParameter.Key is the same as queryParameter.Value.Name?